backends/p4fmt/p4fmt.cpp

```diff
@@ -24,8 +25,21 @@ std::stringstream getFormattedOutput(std::filesystem::path inputFile) {
         return formattedOutput;
     }
 
+    std::unordered_set<const Util::Comment *> globalCommentsList;
+    // get the global list of comments
+    if (!program->objects.empty()) {
+        const auto *firstNode = program->objects.front();
+        if (firstNode->srcInfo.isValid()) {
+            for (const auto *comment : firstNode->srcInfo.getAllFileComments()) {
+                globalCommentsList.insert(comment);
+            }
+        }
+    }
+
+    auto attach = P4::P4Fmt::Attach(globalCommentsList);
+    program = program->apply(attach);
+    // TODO(snapdgn): replace this with `P4Formatter` after rebase
```

Comment:
Do you plan to resolve this TODO in this PR?

Reply:
yes

Reply:
I've squashed some commits and rebased. This now uses P4Formatter.